### ISSUE TITLE
AD Integration - user management - Grouping is not functioning

### ISSUE THREAD

**Issue by dxgundam01** [nextcloud/server#6045]
### Steps to reproduce
1.Sync users from AD to Nextcloud. Success(over 1000)
2.Sync groups from AD to Nextcloud. Success(over 100)
3

### Expected behaviour
Groups filtering must show the assigned users

### Actual behaviour
All groups are empty

### Server configuration

**Operating system**:
centos7
**Web server:**
APACHE
**Database:**
MariaDB
**PHP version:**
php 5.6
**Nextcloud version:** (see Nextcloud admin page)
Nextcloud 12.0.0
**Updated from an older Nextcloud/ownCloud or fresh install:**
 fresh install
**Where did you install Nextcloud from:**

**Signing status:**

No errors have been found.

**List of activated apps:**
[root@localhost html]# sudo -u apache php occ app:list
Enabled:
  - activity: 2.5.2
  - bruteforcesettings: 1.0.2
  - dav: 1.3.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - gallery: 17.0.0
  - groupfolders: 1.1.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - survey_client: 1.0.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - user_external: 0.4
  - user_ldap: 1.2.1
  - user_saml: 1.3.2
  - workflowengine: 1.2.0
Disabled:
  - admin_audit
  - comments
  - encryption
  - files_external
  - firstrunwizard
  - sharebymail
  - systemtags
  - updatenotification
[root@localhost html]# 

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
ActiveDirectory
#### LDAP configuration (delete this part if not used)

<summary>LDAP config</summary>

[root@localhost html]# sudo -u apache php occ ldap:show-config
+-------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
| Configuration                 |                                                                                                                                 |
+-------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                                                                                               |
| hasPagedResultSupport         |                                                                                                                                 |
| homeFolderNamingRule          |                                                                                                                                 |
| lastJpegPhotoLookup           | 0                                                                                                                               |
| ldapAgentName                 | CN=itadmin,OU=IT閮,OU=璐㈠姟琛屾斂浣撶郴,OU=Nubia,DC=nubia,DC=cn                                                                       |
| ldapAgentPassword             | ***                                                                                                                             |
| ldapAttributesForGroupSearch  |                                                                                                                                 |
| ldapAttributesForUserSearch   |                                                                                                                                 |
| ldapBackupHost                |                                                                                                                                 |
| ldapBackupPort                |                                                                                                                                 |
| ldapBase                      | DC=nubia,DC=cn                                                                                                                  |
| ldapBaseGroups                | OU=Nubia,DC=nubia,DC=cn                                                                                                         |
| ldapBaseUsers                 | OU=Nubia,DC=nubia,DC=cn                                                                                                         |
| ldapCacheTTL                  | 600                                                                                                                             |
| ldapConfigurationActive       | 1                                                                                                                               |
| ldapDefaultPPolicyDN          |                                                                                                                                 |
| ldapDynamicGroupMemberURL     |                                                                                                                                 |
| ldapEmailAttribute            | mail                                                                                                                            |
| ldapExperiencedAdmin          | 1                                                                                                                               |
| ldapExpertUUIDGroupAttr       |                                                                                                                                 |
| ldapExpertUUIDUserAttr        |                                                                                                                                 |
| ldapExpertUsernameAttr        | sAMAccountName                                                                                                                  |
| ldapGidNumber                 | gidNumber                                                                                                                       |
| ldapGroupDisplayName          | name                                                                                                                            |
| ldapGroupFilter               | (|(objectClass=organizationalUnit)(objectClass=organization)(objectClass=domain)(objectClass=domainDNS)(objectClass=container)) |
| ldapGroupFilterGroups         |                                                                                                                                 |
| ldapGroupFilterMode           | 1                                                                                                                               |
| ldapGroupFilterObjectclass    |                                                                                                                                 |
| ldapGroupMemberAssocAttr      | member                                                                                                                          |
| ldapHost                      | 10.206.2.6                                                                                                                      |
| ldapIgnoreNamingRules         |                                                                                                                                 |
| ldapLoginFilter               | (&(|(|(memberof=CN=Domain Users,CN=Users,DC=nubia,DC=cn)(primaryGroupID=513)))(samaccountname=%uid))                            |
| ldapLoginFilterAttributes     |                                                                                                                                 |
| ldapLoginFilterEmail          | 0                                                                                                                               |
| ldapLoginFilterMode           | 0                                                                                                                               |
| ldapLoginFilterUsername       | 1                                                                                                                               |
| ldapNestedGroups              | 0                                                                                                                               |
| ldapOverrideMainServer        |                                                                                                                                 |
| ldapPagingSize                | 500                                                                                                                             |
| ldapPort                      | 389                                                                                                                             |
| ldapQuotaAttribute            |                                                                                                                                 |
| ldapQuotaDefault              |                                                                                                                                 |
| ldapTLS                       | 0                                                                                                                               |
| ldapUserDisplayName           | displayname                                                                                                                     |
| ldapUserDisplayName2          |                                                                                                                                 |
| ldapUserFilter                | (objectclass=*)                                                                                                                 |
| ldapUserFilterGroups          |                                                                                                                                 |
| ldapUserFilterMode            | 1                                                                                                                               |
| ldapUserFilterObjectclass     |                                                                                                                                 |
| ldapUuidGroupAttribute        | auto                                                                                                                            |
| ldapUuidUserAttribute         | auto                                                                                                                            |
| turnOffCertCheck              | 0                                                                                                                               |
| turnOnPasswordChange          | 0                                                                                                                               |
| useMemberOfToDetectMembership | 1                                                                                                                               |
+-------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
[root@localhost html]# 

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by LukasReschke** [comment]
cc @nextcloud/ldap

**Comment by blizzz** [comment]
Config looks okay for a typical AD setup. A record of one group and user could help.

It might be that the issue is also solved by https://github.com/nextcloud/server/pull/6453, although a typical AD setup should not be affected. After all, it depends on its configuration/customazation…

**Comment by dnauck** [comment]
@blizzz the fix from #6453 works well, but the group lists (left side panel with groups list) is still empty. In the "All Users" view everything is ok after using patch #6453.

**Comment by skjnldsv** [comment]
As this bug is an old one and ldap received a lot of fixes over the past years, I will close this ticket. If this is still happening please make sure to upgrade to the latest version. After that, feel free to reopen.
